forge-ai/src/main/java/forge/ai/ability/CharmAi.java

```diff
@@ -93,13 +94,22 @@ protected boolean checkApiLogic(Player ai, SpellAbility sa) {
     }
 
     private List<AbilitySub> chooseOptionsAi(List<AbilitySub> choices, final Player ai, boolean isTrigger, int num,
-            int min, boolean allowRepeat) {
+            int min, int pawprintLimit, boolean allowRepeat) {
```

Comment:
i would just pass the SpellAbility sa to this function, so you don't need to have extra pawprintLimit

like i accidently commited there:
https://github.com/Card-Forge/forge/pull/5713/files#diff-fd1617932269fd1af8b5b0c76a9d2597fd611b6c478b39f62bc7baa035d614a6

Reply:
Ah, indeed, done :) :+1: